v2/blockstore/readwrite.go

```diff
@@ -158,36 +176,39 @@ func (b *ReadWrite) resumeWithRoots(roots []cid.Cid) error {
 		// Or the write must have failed before pragma was written.
 		return err
 	}
-	if version != 2 {
-		// The file is not a CARv2 and we cannot resume from it.
+	if !((version == 1 && !v2) || (version == 2 && v2)) {
```

Comment:
I'm having a hard time parsing this in my head. How about:

```
switch {
case version == 1 && !v2:
case version == 2 && v2:
default:
    return fmt.Errorf(...)
}
```

Reply:
I'll have to take you on your word that that's more readable to Go eyes than the original